File: admin/backend/internal/dto/category.go
package dto

import "time"

// CreateCategoryReq is the body of POST /admin/categories.
type CreateCategoryReq struct {
	Name        string `json:"name"        binding:"required,min=1,max=64"`
	Icon        string `json:"icon"        binding:"max=255"`
	Sort        int    `json:"sort"`
	Status      int8   `json:"status"      binding:"omitempty,oneof=1 2"`
	Description string `json:"description" binding:"max=255"`
}

// UpdateCategoryReq is the body of PUT /admin/categories/:id.
type UpdateCategoryReq struct {
	Name        *string `json:"name"        binding:"omitempty,min=1,max=64"`
	Icon        *string `json:"icon"        binding:"omitempty,max=255"`
	Sort        *int    `json:"sort"`
	Status      *int8   `json:"status"      binding:"omitempty,oneof=1 2"`
	Description *string `json:"description" binding:"omitempty,max=255"`
}

// CategoryResp is the wire shape for a single category.
type CategoryResp struct {
	ID          uint64    `json:"id"`
	Name        string    `json:"name"`
	Icon        string    `json:"icon"`
	Sort        int       `json:"sort"`
	Status      int8      `json:"status"`
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

File: admin/frontend/src/api/category.ts
import type { PageData } from '@/utils/request';
import { del, get, post, put } from '@/utils/request';

export interface Category {
  id: number;
  name: string;
  icon: string;
  sort: number;
  status: 1 | 2;
  description: string;
  created_at: string;
  updated_at: string;
}

export interface CreateCategoryReq {
  name: string;
  icon?: string;
  sort?: number;
  status?: 1 | 2;
  description?: string;
}

export interface UpdateCategoryReq {
  name?: string;
  icon?: string;
  sort?: number;
  status?: 1 | 2;
  description?: string;
}

function qs(params: Record<string, unknown>): string {
  const usp = new URLSearchParams();
  for (const [k, v] of Object.entries(params)) {
    if (v !== undefined && v !== null && v !== '') usp.set(k, String(v));
  }
  return usp.toString();
}

export const categoryApi = {
  list: (status?: 1 | 2) => get<Category[]>(`/admin/categories?${qs({ status: status ?? '' })}`),
  create: (body: CreateCategoryReq) => post<Category>('/admin/categories', body),
  update: (id: number, body: UpdateCategoryReq) => put<Category>(`/admin/categories/${id}`, body),
  remove: (id: number) => del(`/admin/categories/${id}`),
};
